src/main/java/uk/gov/hmcts/ccd/service/MessagePublisherRunnable.java

```diff
@@ -63,10 +65,31 @@ private void processUnpublishedMessages(Pageable pageable,
         }
     }
 
+    private String getProperty(JsonNode data, String header) {
+        try {
+            return data.get(header).asText();
```

Comment:
rather than using exceptions, does the Jakson API provide a method that allows to test if a JSON contains a property?